qiskit_aer/quantum_info/states/aer_statevector.py

```diff
@@ -0,0 +1,273 @@
+            AerError: if input data is not valid.
+
+        Additional Information:
+            The ``dims`` kwarg is used to ``Statevector`` constructor.
+
+        """
+        if 'dims' in configs:
+            dims = configs.pop('dims')
+        else:
+            dims = None
+
+        if isinstance(data, (QuantumCircuit, Instruction)):
+            data, aer_state = AerStatevector._from_instruction(data, None, **configs)
+        elif isinstance(data, list):
+            data, aer_state = AerStatevector._from_ndarray(np.array(data, dtype=complex), **configs)
+        elif isinstance(data, np.ndarray):
+            data, aer_state = AerStatevector._from_ndarray(data, **configs)
+        elif isinstance(data, AerStatevector):
+            aer_state = data._aer_state
+            if dims is None:
+                dims = data._op_shape._dims_l
+            data = data._data
+        else:
+            raise AerError(f'Input data is not supported: type={data.__class__}, data={data}')
+
+        super().__init__(data, dims=dims)
+        self._aer_state = aer_state
+        self._result = None
+        self._configs = configs
+
+    def _last_result(self):
+        if self._result is None:
+            self._result = self._aer_state.last_result()
+        return self._result
+
+    @property
+    def metadata(self):
+        """Return result metadata of an operation that executed lastly."""
+        if self._last_result() is None:
+            raise AerError('AerState was not used and metdata does not exist.')
+        return self._last_result()['metadata']
+
+    def __deepcopy__(self, _memo=None):
+        data, aer_state = AerStatevector._from_instruction(
+            QuantumCircuit(self._aer_state.num_qubits),
+            self._data, **self._configs)
+        return AerStatevector(data, _aer_state=aer_state, **self._configs)
+
+    def __eq__(self, other):
+        eq_quantumqtate = isinstance(other, Statevector) and self.dims() == other.dims()
+        eq_statevector = np.allclose(self._data, other._data, rtol=self.rtol, atol=self.atol)
+        return eq_quantumqtate and eq_statevector
+
+    def evolve(self, other, qargs=None):
+        """Evolve a quantum state by the operator.
+
+        Args:
+            other (Operator): The operator to evolve by.
+            qargs (list): a list of Statevector subsystem positions to apply
+                           the operator on.
+
+        Returns:
+            Statevector: the output quantum state.
+
+        Raises:
+            AerError: if the operator dimension does not match the
+                         specified Statevector subsystem dimensions.
+        """
+        if qargs is None:
+            qargs = getattr(other, "qargs", None)
+
+        if isinstance(other, (QuantumCircuit, Instruction)):
+            data, aer_state = AerStatevector._from_instruction(other, self._data, **self._configs)
+            return AerStatevector(data, _aer_state=aer_state, **self._configs)
```

Comment:
This passes `_aer_state` as an argument, but that's not interpreted by the actual `__init__` method (see the top review comment for a potentially simpler method here).

Reply:
I accidentally deleted `_aer_state` argument in `__init__`. It is private and used by only `AerStatevector`.